src/features/embalm/stepContent/hooks/useCreateSarcophagus/useArchaeologistSignatureNegotiation.ts

```diff
@@ -55,8 +59,19 @@ export function useArchaeologistSignatureNegotiation() {
 
     await Promise.all(
       selectedArchaeologists.map(async arch => {
-        if (!arch.connection)
-          throw new Error(`No connection to archaeologist ${JSON.stringify(arch)}`);
+        if (!arch.connection) {
+          console.log(`${arch.profile.peerId} connection is undefined`);
+          setArchaeologistException(arch.profile.peerId, {
+            code: ArchaeologistExceptionCode.CONNECTION_EXCEPTION,
+            message: 'No connection to archaeologist',
+          });
+
+          if (arch.fullPeerId) {
+            await dialSelectedArchaeologists();
+          }
+
+          return;
+        }
```

Comment:
@DarksightKellar -- wondering if instead of doing this routine inside of this `Promise.all` --

1. Let this routine fail
2. On retry, re-dial selected archaeologists prior to running the negotiation.

We would need to know that this is a retry on the negotiation step to do this.

Reply:
Hmm I'd like to explore this further. It did cross my mind but didn't want to mess with the retry mechanism just yet. But if we could refactor retry to do more than just re-run the current step, and have more context, we could be more intentional about *what* it is we're retrying, and how